nondex-maven-plugin/src/main/java/edu/illinois/nondex/plugin/NonDexSurefireExecution.java

```diff
@@ -79,9 +82,11 @@ protected void setupArgline() {
             configElement.getChild("test").setValue(this.configuration.testName);
         }
         Logger.getGlobal().log(Level.FINE, "Running surefire with: " + this.configuration.toArgLine());
+        Logger.getGlobal().log(Level.SEVERE, mavenRepoLocal);
         this.mavenProject.getProperties().setProperty("argLine",
-                "" + "-Xbootclasspath/p:" + pathToNondex + " " + this.originalArgLine + " "
-                + this.configuration.toArgLine());
+                "" + "-Xbootclasspath/p:" + pathToNondex + ":" + Paths.get(mavenRepoLocal,
+                        "edu/illinois/nondex-maven-plugin/1.0.2-SNAPSHOT/nondex-maven-plugin-1.0.2-SNAPSHOT.jar")
```

Comment:
This should not hard-code 1.0.2-SNAPSHOT. The version will change, and it needs basically the current version from: Configuration.VERSION. That is incremented on every release.